src/main/java/com/melon/portfoliomanager/services/StockPricesService.java

```diff
@@ -9,22 +9,30 @@
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
+import java.util.Map;
+
 @Service
 public class StockPricesService {
     private static final Logger logger = LoggerFactory.getLogger(StockPricesService.class);
     private final CompanyStocksManager companyStocksManager = new CompanyStocksManager();
     private final StockPricesHttpService stockPricesHttpService;
+    private final MessageBrokerService messageBrokerService;
 
     @Autowired
-    public StockPricesService(@Qualifier("stock-prices-mock-api") StockPricesHttpService stockPricesHttpService) {
+    public StockPricesService(@Qualifier("stock-prices-mock-api") StockPricesHttpService stockPricesHttpService,
+                              @Qualifier("kafka") MessageBrokerService messageBrokerService) {
         this.stockPricesHttpService = stockPricesHttpService;
+        this.messageBrokerService = messageBrokerService;
     }
 
     @Scheduled(fixedDelay = 15 * 60 * 1000)
     private void fetchCompanyStocks() {
         try {
             ResponseEntity<?> stockPricesResponse = stockPricesHttpService.getStockPrices();
             StockPricesDTO latestCompanyStocks = (StockPricesDTO) stockPricesResponse.getBody();
+            Map<String, Double> companyStocksWithPriceChanges =
+                    companyStocksManager.getCompanyStocksWithExtremePriceChanges(latestCompanyStocks.getStockPrices());
+            messageBrokerService.sendMessage(companyStocksWithPriceChanges);
```

Comment:
Check if the list is empty. We shouldn't be sending empty messages to the queue.